src/super_gradients/training/metrics/pose_estimation_metrics.py

```diff
@@ -133,98 +134,128 @@ def reset(self) -> None:
+
+        :param gt_iscrowd:  Optional argument indicating which instance is annotated with `iscrowd` flog and is not used for evaluation;
+                            If not provided, all instances are considered as non-crowd targets.
+                            For instance, in CrowdPose all instances are considered as "non-crowd".
+
+        :param gt_bboxes:   Bounding boxes of the groundtruth instances (XYWH).
+                            This is COCO-specific and is used in OKS computation for instances w/o visible keypoints.
+                            If not provided, the bounding box is computed as the minimum bounding box that contains all visible keypoints.
+
+        :param gt_areas:    Area of the groundtruth area. in COCO this is the area of the corresponding segmentation mask and not the bounding box,
+                            so it cannot be computed programmatically. It's value used in object-keypoint similarity metric (OKS) computation.
+                            If not provided, the area is computed as the product of the width and height of the bounding box.
+                            (For instance this is used in CrowdPose dataset)
+        :param gt_samples:  List of ground-truth samples
 
         """
-        predicted_poses, predicted_scores = self.post_prediction_callback(preds)  # Decode raw predictions into poses
+        predictions: List[PoseEstimationPredictions] = self.post_prediction_callback(preds)  # Decode raw predictions into poses
 
-        if gt_bboxes is None:
-            gt_bboxes = [compute_visible_bbox_xywh(torch.tensor(joints[:, :, 0:2]), torch.tensor(joints[:, :, 2])) for joints in gt_joints]
-
-        if gt_areas is None:
-            gt_areas = [bboxes[:, 2] * bboxes[:, 3] for bboxes in gt_bboxes]
+        if gt_samples is not None:
+            self._update_with_samples(predictions, gt_samples)
+        else:
+            self._update_with_old_style_args(predictions, gt_joints, gt_bboxes, gt_areas, gt_iscrowd)
 
-        if gt_iscrowd is None:
-            gt_iscrowd = [[False] * len(x) for x in gt_joints]
+    def _update_with_samples(self, predictions: List[PoseEstimationPredictions], gt_samples: List[PoseEstimationSample]):
+        for i in range(len(predictions)):
+            self.update_single_image(
+                predicted_poses=predictions[i].poses,
+                predicted_scores=predictions[i].scores,
+                gt_joints=gt_samples[i].joints,
+                gt_bboxes=gt_samples[i].bboxes,
+                gt_areas=gt_samples[i].areas,
+                gt_iscrowd=gt_samples[i].is_crowd,
+            )
 
-        for i in range(len(predicted_poses)):
+    def _update_with_old_style_args(
+        self,
+        predictions: List[PoseEstimationPredictions],
+        gt_joints: List[np.ndarray],
+        gt_bboxes: Optional[List[np.ndarray]],
+        gt_areas: Optional[List[np.ndarray]],
+        gt_iscrowd: Optional[List[np.ndarray]],
+    ):
```

Comment:
Maybe write docstrings explaining that this is out of backward compatibility to change introduced in version XXX?

Reply:
Fixed in d27f6245b5687c087a02e5143d760b9aa9ad2c06